Panic running docker-machine ls

I'm running on OSX EL Capitain, after starting my virtual dev machine with `docker-machine` command, I wan't able to configure my shell and below the error:

```
Error running connection boilerplate: Error getting driver URL: Something went wrong running an SSH command!
command : ip addr show dev eth1
err     : exit status 255
output  :
```

When I run `docker-machine ls`, I got:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0xad9d2]

goroutine 1 [running]:
github.com/docker/machine/commands.cmdLs(0xc820118360, 0x0, 0x0)
	/go/src/github.com/docker/machine/commands/ls.go:95 +0xa52
github.com/docker/machine/commands.fatalOnError.func1(0xc820118360)
	/go/src/github.com/docker/machine/commands/commands.go:44 +0x29
github.com/docker/machine/cli.Command.Run(0x5563a0, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0x55e030, 0xd, 0x0, ...)
	/go/src/github.com/docker/machine/cli/command.go:137 +0x1055
github.com/docker/machine/cli.(*App).Run(0xc820118120, 0xc82000a1c0, 0x2, 0x2, 0x0, 0x0)
	/go/src/github.com/docker/machine/cli/app.go:175 +0xfd6
main.main()
	/go/src/github.com/docker/machine/cmd/machine.go:144 +0xcc3

goroutine 5 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 17 [select]:
github.com/docker/machine/libmachine/drivers/plugin/localbinary.(*LocalBinaryPlugin).execServer(0xc820109380, 0x0, 0x0)
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:194 +0xc86
github.com/docker/machine/libmachine/drivers/plugin/localbinary.(*LocalBinaryPlugin).Serve(0xc820109380, 0x0, 0x0)
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:211 +0x2d
github.com/docker/machine/libmachine/drivers/rpc.NewRpcClientDriver.func1(0xc820109380)
	/go/src/github.com/docker/machine/libmachine/drivers/rpc/client_driver.go:58 +0x25
created by github.com/docker/machine/libmachine/drivers/rpc.NewRpcClientDriver
	/go/src/github.com/docker/machine/libmachine/drivers/rpc/client_driver.go:63 +0xcf

goroutine 19 [select]:
github.com/docker/machine/libmachine/drivers/plugin/localbinary.stream(0xc820112280, 0xc820128360, 0xc8201283c0)
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:154 +0x37d
created by github.com/docker/machine/libmachine/drivers/plugin/localbinary.(*LocalBinaryPlugin).AttachStream
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:170 +0x8e

goroutine 20 [select]:
github.com/docker/machine/libmachine/drivers/plugin/localbinary.stream(0xc820112300, 0xc820128420, 0xc820128480)
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:154 +0x37d
created by github.com/docker/machine/libmachine/drivers/plugin/localbinary.(*LocalBinaryPlugin).AttachStream
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:170 +0x8e

goroutine 21 [syscall]:
syscall.Syscall(0x3, 0x5, 0xc820145000, 0x1000, 0xd83, 0x0, 0x0)
	/usr/local/go/src/syscall/asm_darwin_amd64.s:16 +0x5
syscall.read(0x5, 0xc820145000, 0x1000, 0x1000, 0xd83, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_darwin_amd64.go:972 +0x5f
syscall.Read(0x5, 0xc820145000, 0x1000, 0x1000, 0xd83, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:160 +0x4d
os.(*File).read(0xc8201040b8, 0xc820145000, 0x1000, 0x1000, 0x300d07, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:211 +0x75
os.(*File).Read(0xc8201040b8, 0xc820145000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/file.go:95 +0x8a
bufio.(*Scanner).Scan(0xc820112300, 0xc8201284e0)
	/usr/local/go/src/bufio/scan.go:180 +0x877
github.com/docker/machine/libmachine/drivers/plugin/localbinary.stream.func1(0xc820112300, 0xc8201284e0)
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:149 +0x25
created by github.com/docker/machine/libmachine/drivers/plugin/localbinary.stream
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:152 +0x81

goroutine 7 [syscall]:
syscall.Syscall(0x3, 0x3, 0xc820144010, 0xff0, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/asm_darwin_amd64.s:16 +0x5
syscall.read(0x3, 0xc820144010, 0xff0, 0xff0, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_darwin_amd64.go:972 +0x5f
syscall.Read(0x3, 0xc820144010, 0xff0, 0xff0, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:160 +0x4d
os.(*File).read(0xc8201040a8, 0xc820144010, 0xff0, 0xff0, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:211 +0x75
os.(*File).Read(0xc8201040a8, 0xc820144010, 0xff0, 0xff0, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/file.go:95 +0x8a
bufio.(*Scanner).Scan(0xc820112280, 0x0)
	/usr/local/go/src/bufio/scan.go:180 +0x877
github.com/docker/machine/libmachine/drivers/plugin/localbinary.stream.func1(0xc820112280, 0xc820018240)
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:149 +0x25
created by github.com/docker/machine/libmachine/drivers/plugin/localbinary.stream
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:152 +0x81

goroutine 33 [IO wait]:
net.runtime_pollWait(0xacf000, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820156060, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820156060, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820156000, 0xc820166000, 0x1000, 0x1000, 0x0, 0xa4e050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc82015c000, 0xc820166000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
bufio.(*Reader).fill(0xc82015e060)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Read(0xc82015e060, 0xc820148040, 0x1, 0x9, 0x6377c0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:207 +0x260
io.ReadAtLeast(0xacf138, 0xc82015e060, 0xc820148040, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:298 +0xe6
io.ReadFull(0xacf138, 0xc82015e060, 0xc820148040, 0x1, 0x9, 0xc8201540f0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:316 +0x62
encoding/gob.decodeUintReader(0xacf138, 0xc82015e060, 0xc820148040, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
	/usr/local/go/src/encoding/gob/decode.go:121 +0x92
encoding/gob.(*Decoder).recvMessage(0xc820154080, 0xc82010dbc0)
	/usr/local/go/src/encoding/gob/decoder.go:76 +0x5e
encoding/gob.(*Decoder).decodeTypeSequence(0xc820154080, 0x638600, 0xc820154080)
	/usr/local/go/src/encoding/gob/decoder.go:140 +0x47
encoding/gob.(*Decoder).DecodeValue(0xc820154080, 0x3cfdc0, 0xc820188000, 0x16, 0x0, 0x0)
	/usr/local/go/src/encoding/gob/decoder.go:208 +0x15d
encoding/gob.(*Decoder).Decode(0xc820154080, 0x3cfdc0, 0xc820188000, 0x0, 0x0)
	/usr/local/go/src/encoding/gob/decoder.go:185 +0x289
net/rpc.(*gobClientCodec).ReadResponseHeader(0xc8201521b0, 0xc820188000, 0x0, 0x0)
	/usr/local/go/src/net/rpc/client.go:223 +0x51
net/rpc.(*Client).input(0xc82015e120)
	/usr/local/go/src/net/rpc/client.go:109 +0xbf
created by net/rpc.NewClientWithCodec
	/usr/local/go/src/net/rpc/client.go:201 +0xd2

goroutine 34 [sleep]:
time.Sleep(0xbebc200)
	/usr/local/go/src/runtime/time.go:59 +0xf9
github.com/docker/machine/libmachine/drivers/rpc.NewRpcClientDriver.func2(0xc820158100)
	/go/src/github.com/docker/machine/libmachine/drivers/rpc/client_driver.go:91 +0x1dc
created by github.com/docker/machine/libmachine/drivers/rpc.NewRpcClientDriver
	/go/src/github.com/docker/machine/libmachine/drivers/rpc/client_driver.go:94 +0x38a

goroutine 49 [select]:
github.com/docker/machine/libmachine/drivers/plugin/localbinary.(*LocalBinaryPlugin).execServer(0xc82018a580, 0x0, 0x0)
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:194 +0xc86
github.com/docker/machine/libmachine/drivers/plugin/localbinary.(*LocalBinaryPlugin).Serve(0xc82018a580, 0x0, 0x0)
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:211 +0x2d
github.com/docker/machine/libmachine/drivers/rpc.NewRpcClientDriver.func1(0xc82018a580)
	/go/src/github.com/docker/machine/libmachine/drivers/rpc/client_driver.go:58 +0x25
created by github.com/docker/machine/libmachine/drivers/rpc.NewRpcClientDriver
	/go/src/github.com/docker/machine/libmachine/drivers/rpc/client_driver.go:63 +0xcf

goroutine 50 [select]:
github.com/docker/machine/libmachine/drivers/plugin/localbinary.stream(0xc8201be000, 0xc8201b0420, 0xc8201b0480)
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:154 +0x37d
created by github.com/docker/machine/libmachine/drivers/plugin/localbinary.(*LocalBinaryPlugin).AttachStream
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:170 +0x8e

goroutine 51 [select]:
github.com/docker/machine/libmachine/drivers/plugin/localbinary.stream(0xc8201be080, 0xc8201b04e0, 0xc8201b0540)
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:154 +0x37d
created by github.com/docker/machine/libmachine/drivers/plugin/localbinary.(*LocalBinaryPlugin).AttachStream
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:170 +0x8e

goroutine 52 [syscall]:
syscall.Syscall(0x3, 0x9, 0xc8201bd000, 0x1000, 0xbc8, 0x0, 0x0)
	/usr/local/go/src/syscall/asm_darwin_amd64.s:16 +0x5
syscall.read(0x9, 0xc8201bd000, 0x1000, 0x1000, 0xbc8, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_darwin_amd64.go:972 +0x5f
syscall.Read(0x9, 0xc8201bd000, 0x1000, 0x1000, 0xbc8, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:160 +0x4d
os.(*File).read(0xc8201a4158, 0xc8201bd000, 0x1000, 0x1000, 0x300d07, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:211 +0x75
os.(*File).Read(0xc8201a4158, 0xc8201bd000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/file.go:95 +0x8a
bufio.(*Scanner).Scan(0xc8201be080, 0xc8201b05a0)
	/usr/local/go/src/bufio/scan.go:180 +0x877
github.com/docker/machine/libmachine/drivers/plugin/localbinary.stream.func1(0xc8201be080, 0xc8201b05a0)
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:149 +0x25
created by github.com/docker/machine/libmachine/drivers/plugin/localbinary.stream
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:152 +0x81

goroutine 35 [syscall]:
syscall.Syscall(0x3, 0x7, 0xc8201bc010, 0xff0, 0xc82015e138, 0x328142, 0xc820168000)
	/usr/local/go/src/syscall/asm_darwin_amd64.s:16 +0x5
syscall.read(0x7, 0xc8201bc010, 0xff0, 0xff0, 0x3cfd60, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_darwin_amd64.go:972 +0x5f
syscall.Read(0x7, 0xc8201bc010, 0xff0, 0xff0, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:160 +0x4d
os.(*File).read(0xc8201a4148, 0xc8201bc010, 0xff0, 0xff0, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:211 +0x75
os.(*File).Read(0xc8201a4148, 0xc8201bc010, 0xff0, 0xff0, 0xc820186140, 0x0, 0x0)
	/usr/local/go/src/os/file.go:95 +0x8a
bufio.(*Scanner).Scan(0xc8201be000, 0x0)
	/usr/local/go/src/bufio/scan.go:180 +0x877
github.com/docker/machine/libmachine/drivers/plugin/localbinary.stream.func1(0xc8201be000, 0xc82015e660)
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:149 +0x25
created by github.com/docker/machine/libmachine/drivers/plugin/localbinary.stream
	/go/src/github.com/docker/machine/libmachine/drivers/plugin/localbinary/plugin.go:152 +0x81

goroutine 8 [IO wait]:
net.runtime_pollWait(0xacef40, 0x72, 0xc8200121a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201560d0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201560d0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820156070, 0xc8201d1000, 0x1000, 0x1000, 0x0, 0xa4e050, 0xc8200121a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc82002a0d8, 0xc8201d1000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
bufio.(*Reader).fill(0xc820018300)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Read(0xc820018300, 0xc8200135f0, 0x1, 0x9, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:207 +0x260
io.ReadAtLeast(0xacf138, 0xc820018300, 0xc8200135f0, 0x1, 0x9, 0x1, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:298 +0xe6
io.ReadFull(0xacf138, 0xc820018300, 0xc8200135f0, 0x1, 0x9, 0xc8200882f0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:316 +0x62
encoding/gob.decodeUintReader(0xacf138, 0xc820018300, 0xc8200135f0, 0x9, 0x9, 0x0, 0x1, 0x0, 0x0)
	/usr/local/go/src/encoding/gob/decode.go:121 +0x92
encoding/gob.(*Decoder).recvMessage(0xc820088280, 0xc8201ebbc0)
	/usr/local/go/src/encoding/gob/decoder.go:76 +0x5e
encoding/gob.(*Decoder).decodeTypeSequence(0xc820088280, 0x638600, 0xc820088280)
	/usr/local/go/src/encoding/gob/decoder.go:140 +0x47
encoding/gob.(*Decoder).DecodeValue(0xc820088280, 0x3cfdc0, 0xc8201ce1b0, 0x16, 0x0, 0x0)
	/usr/local/go/src/encoding/gob/decoder.go:208 +0x15d
encoding/gob.(*Decoder).Decode(0xc820088280, 0x3cfdc0, 0xc8201ce1b0, 0x0, 0x0)
	/usr/local/go/src/encoding/gob/decoder.go:185 +0x289
net/rpc.(*gobClientCodec).ReadResponseHeader(0xc8201ce150, 0xc8201ce1b0, 0x0, 0x0)
	/usr/local/go/src/net/rpc/client.go:223 +0x51
net/rpc.(*Client).input(0xc8200183c0)
	/usr/local/go/src/net/rpc/client.go:109 +0xbf
created by net/rpc.NewClientWithCodec
	/usr/local/go/src/net/rpc/client.go:201 +0xd2

goroutine 9 [sleep]:
time.Sleep(0xbebc200)
	/usr/local/go/src/runtime/time.go:59 +0xf9
github.com/docker/machine/libmachine/drivers/rpc.NewRpcClientDriver.func2(0xc82000baa0)
	/go/src/github.com/docker/machine/libmachine/drivers/rpc/client_driver.go:91 +0x1dc
created by github.com/docker/machine/libmachine/drivers/rpc.NewRpcClientDriver
	/go/src/github.com/docker/machine/libmachine/drivers/rpc/client_driver.go:94 +0x38a

goroutine 10 [chan receive]:
github.com/docker/machine/commands.attemptGetHostState(0xc8201280c0, 0xc820018540)
	/go/src/github.com/docker/machine/commands/ls.go:260 +0x158
created by github.com/docker/machine/commands.getHostState
	/go/src/github.com/docker/machine/commands/ls.go:286 +0x6f

goroutine 12 [chan receive]:
net/rpc.(*Client).Call(0xc82015e120, 0x5a6020, 0x16, 0x3e9a40, 0x808728, 0x3d3480, 0xc8200136a0, 0x0, 0x0)
	/usr/local/go/src/net/rpc/client.go:315 +0xcd
github.com/docker/machine/libmachine/drivers/rpc.(*InternalClient).Call(0xc8201580e0, 0x5a6020, 0x16, 0x3e9a40, 0x808728, 0x3d3480, 0xc8200136a0, 0x0, 0x0)
	/go/src/github.com/docker/machine/libmachine/drivers/rpc/client_driver.go:40 +0xfd
github.com/docker/machine/libmachine/drivers/rpc.(*RpcClientDriver).rpcStringCall(0xc820158100, 0x5a6020, 0x16, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/machine/libmachine/drivers/rpc/client_driver.go:148 +0xe0
github.com/docker/machine/libmachine/drivers/rpc.(*RpcClientDriver).GetURL(0xc820158100, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/machine/libmachine/drivers/rpc/client_driver.go:193 +0x4e
github.com/docker/machine/libmachine/host.(*Host).GetURL(0xc8201280c0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/machine/libmachine/host/host.go:149 +0x52
github.com/docker/machine/commands.attemptGetHostState.func2(0xc8201280c0, 0xc820018660)
	/go/src/github.com/docker/machine/commands/ls.go:247 +0x52
created by github.com/docker/machine/commands.attemptGetHostState
	/go/src/github.com/docker/machine/commands/ls.go:257 +0xe8
```

--- a/commands/ls.go
+++ b/commands/ls.go
@@ -91,7 +91,7 @@ func cmdLs(c CommandLine) error {
 
 		swarmInfo := ""
 
-		if item.SwarmOptions.Discovery != "" {
+		if item.SwarmOptions != nil && item.SwarmOptions.Discovery != "" {
 			swarmInfo = swarmMasters[item.SwarmOptions.Discovery]
 			if item.SwarmOptions.Master {
 				swarmInfo = fmt.Sprintf("%s (master)", swarmInfo)

